discovery-infra/tests/base_test.py

```diff
@@ -90,12 +87,45 @@ def assert_string_length(string, expected_len):
         assert len(string) == expected_len, "Expected len string of: " + str(expected_len) + \
                                             " rather than: " + str(len(string)) + " String value: " + string
 
-    @staticmethod
-    def collect_test_logs(cluster, api_client, test):
+    def collect_test_logs(self, cluster, api_client, test, nodes):
         log_dir_name = f"{env_variables['log_folder']}/{test.name}"
         with suppress(ApiException):
             cluster_details = json.loads(json.dumps(cluster.get_details().to_dict(), sort_keys=True, default=str))
             download_logs(api_client, cluster_details, log_dir_name, test.result_call.failed)
+        if test.result_call.failed:
+            self._collect_virsh_logs(nodes, log_dir_name)
+
+    @staticmethod
+    def _collect_virsh_logs(nodes, log_dir_name):
+        logging.info('---TEARDOWN --- Collecting virsh logs\n')
+        if not os.path.exists(log_dir_name):
+            logging.info(f"Creating virsh logs directory: {log_dir_name}")
```

Comment:
I think this log is redundant.

Reply:
It is the directory that can be created at collect_test_logs method, but, this folder created only if cluster obj exist. Not in all failures cluster obj created, so for that cases im adding folder creation.

Reply:
I still don't understand why it is an important log for the log collection workflow.
Especially in case `exist_ok=True` would be set. Already created/newly created - it doesn't really matter.